tests/shared/sdk/usbboot.spec.js

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2017 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const m = require('mochainon')
+const _ = require('lodash')
+const usb = require('usb')
+const usbboot = require('../../../lib/shared/sdk/usbboot')
+
+const createUSBDeviceStub = (options) => {
+  return {
+    deviceDescriptor: {
+      idVendor: options.vendorID,
+      idProduct: options.productID,
+      iManufacturer: options.manufacturerIndex,
+      iProduct: options.productIndex
+    },
+    _configDescriptor: {
+      bNumInterfaces: options.interfaces
+    },
+    open: _.noop,
+    close: _.noop,
+    getStringDescriptor: (index, callback) => {
+      callback(null, options.strings[index])
+    }
+  }
+}
+
+const BCM2835_STUB = createUSBDeviceStub({
```

Comment:
Just for the sake of completeness, probably wouldn't hurt to also add a `BCM2837_STUB` ?